sec1_intro_and_setup/1.1_basics.rb

```diff
@@ -0,0 +1,14 @@
+# Section 1.1
+# Let's code the quiz!
+
+address = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+# print array
+p address
+
+# print array with index 5
+p address[5]
+
+# reverse the array
+new_address = address.reverse!
```

Comment:
I would think that it save memory because it reuse the variable.

abc.reverse! <= modify abc
abc.reverse <= return the value

returning the value means you need to assign to another variable.
Official definition of ruby of collection is "any class that represent a collection of values". To make it simply, think of it like array & hash as a collection.

To choose which one you use, it depends on how your program flow. Design & code for humans first then only for machines unless it becomes a problem.

---

On that note, IF you don't need `new_address`, then don't use it. If you want to make a new name, then don't use `!`.